File: python/arrays/missingRepeated.py
def findMissingAndRepeatedValues(grid: list[list[int]]) -> list[int]:
    n = len(grid)
    hash = [0] * (n * n + 1)

    for i in range(0, n):
        for j in range(0, n):
            hash[grid[i][j]] += 1

    missing = -1
    repeated = -1
    for i in range(1, len(hash)):
        if hash[i] == 0:
            missing = i
        elif hash[i] > 1:
            repeated = i

    return [repeated, missing]


grid = [[9, 1, 7], [8, 9, 2], [3, 4, 6]]
print(findMissingAndRepeatedValues(grid))

File: cpp/missingRepeated.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> findMissingAndRepeatedValues(vector<vector<int>> &grid) {
  int n = grid.size();
  vector<int> ans(2, 0);
  vector<int> hash(n * n + 1, 0);

  for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
      hash[grid[i][j]]++;
    }
  }

  for (int i = 0; i < hash.size(); i++) {
    if (hash[i] == 0)
      ans[1] = i;
    else if (hash[i] == 2)
      ans[0] = i;
  }

  return ans;
}

int main() {

  vector<vector<int>> grid = {{1, 3}, {2, 2}};

  vector<int> ans = findMissingAndRepeatedValues(grid);

  for (int ele : ans) {
    cout << ele << " ";
  }
}
